paimon-trino-common/src/main/java/org/apache/paimon/trino/TrinoMetadata.java

```diff
@@ -448,7 +461,37 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(
                                 .intersect(TupleDomain.withColumnDomains(unenforcedDomains));
 
         return Optional.of(
-                new ConstraintApplicationResult<>(trinoTableHandle.copy(newFilter), remain, false));
+                new ConstraintApplicationResult<>(
+                        trinoTableHandle.copy(newFilter, expressionFilter), remain, false));
+    }
+
+    private static TupleDomain<TrinoColumnHandle> getTrinoColumnHandleTupleDomain(
+            Constraint constraint) {
+        Map<TrinoColumnHandle, Domain> expressionPredicates = new HashMap<>();
+
+        if (constraint.getExpression() instanceof Call) {
+            Call expression = (Call) constraint.getExpression();
+            Map<String, ColumnHandle> assignments = constraint.getAssignments();
+
+            switch (expression.getFunctionName().getName()) {
+                case "$equal":
```

Comment:
Can we put there in static final string?

private static final String TRINO_EQUAL = "$equal"